File: lib/cell.rb
class Cell
  def initialize(state = :alive, x = 0, y = 0)
    @alive = true unless state == :dead
    @x = x
    @y = y
  end
  
  attr_accessor :x
  attr_accessor :y

  def alive?
    @alive
  end

  def to_s
    alive? ? "#" : "-"
  end
end

File: scala/src/main/scala/Cell.scala
package gol

import gol.TypeAliases._

class Cell(val living: Boolean, val coord: Coordinate) {
  var isLiving = living;
  def dead = !isLiving
  def alive = isLiving
  def live() { isLiving = true }
  def kill() { isLiving = false }
}
